Modules/Core/Common/include/itkMetaDataObject.hxx

```diff
@@ -28,9 +28,79 @@
 #ifndef itkMetaDataObject_hxx
 #define itkMetaDataObject_hxx
 
+#include <type_traits>
+#include <iterator>
+
+template <typename T, typename = void>
+inline constexpr bool is_iterable_print_v = false;
+// Specialize for std::vector<T> and std::vector<std::vector<T>>
+template <typename T>
+inline constexpr bool is_iterable_print_v<std::vector<T>, std::void_t<>> = true;
+template <typename T>
+inline constexpr bool is_iterable_print_v<std::vector<std::vector<T>>, std::void_t<>> = true;
 
 namespace itk
 {
+template <typename TIterable>
+void
+printIterable(std::ostream & os, const TIterable & iterable)
```

Comment:
Interesting pull request, thanks @phcerdan 

Can you possibly hide `printIterable` (or whatever its name will be) from the public API? Because it isn't intended to be exposed directly to the end user, right? I guess it should be possible to make it a private (preferably `static`) member function of `MetaDataObject`. Otherwise, it might be possible to define it as a lambda, hidden locally inside `MetaDataObject::PrintValue`, but that seems more challenging to do.

Reply:
If you can make it private, you might as well just name the function something like `PrintHelper`, rather than `printIterable`.  My 2 cent!